add NcML previewer #17

pdurbin opened the pull request:
- Closes #17

Here's a screenshot of how it looks:

![Screen Shot 2022-12-20 at 2 26 47 PM](https://user-images.githubusercontent.com/21006/208753106-886ae7a0-2731-4c74-b8d5-9d11023e20a2.png)

qqmyers reviewed (changes requested):
Didn't run this but it follows the basic pattern of all the other ones. I left a couple requests and a question about XSS filtering.

pdurbin commented on 5.2curlcommands.md:
```diff
@@ -579,3 +579,65 @@ curl -X POST -H 'Content-type: application/json' http://localhost:8080/api/admin
+  "toolName":"ncmlPreviewer",
+  "scope":"file",
+  "types":["preview"],
+  "toolUrl":"https://gdcc.github.io/dataverse-previewers/previewers/v1.3/NcmlPreview.html",
```
Sounds good. As discussed I [updated the wiki](https://github.com/gdcc/dataverse-previewers/wiki/How-to-create-a-previewer/_compare/266efc11c04182c039315836b633d3860d568dd9...3829a7cd26364b16511ca4d7c70c9f96be642349) and will tweak the README.

pdurbin commented on previewers/v1.3/js/ncml.js:
```diff
@@ -0,0 +1,15 @@
+    $( '.ncmlPreviewText' ).text( ncmlPreviewText );
+}
+
+function writeContentAndData(data, fileUrl, file, title, authors) {
```
Done in a0ee97c.

pdurbin commented on previewers/v1.3/js/ncml.js:
```diff
@@ -0,0 +1,15 @@
+
+function writeContentAndData(data, fileUrl, file, title, authors) {
+    addStandardPreviewHeader(file, title, authors);
+    $('.preview').append($("<textarea/>").html(data)
```
Sure, I'm not exactly sure how it works but I copied it from text.js to ncml.js in 955ca61 and the XML still renders. Please take a look.

pdurbin commented:
Because I can't put this PR on https://github.com/orgs/IQSS/projects/2 I'm using the following issue to represent its state:

- https://github.com/IQSS/dataverse/issues/9242

I'm about to move it back to review since I believe I've addressed the feedback so far.

While I'm here I'll leave a screenshot of the expected error page, which I was just testing:

![Screen Shot 2022-12-22 at 11 46 13 AM](https://user-images.githubusercontent.com/21006/209189118-5c581ec8-dc3b-4942-89ab-4700aa31b0a3.png)

qqmyers commented:
Looks good. The XSS filtering might be more than needed (i.e. the html previewer filters much less than the text previewer), but that can be adjusted as people play with real NetCDF files and decide what's useful.
